pkg/sdk/go/users_test.go

```diff
@@ -128,6 +128,165 @@ func TestCreateUser(t *testing.T) {
+	mockAuthzDB[user.Email] = append(mockAuthzDB[user.Email], mocks.SubjectSet{Object: "authorities", Relation: "member"})
+	auth := mocks.NewAuthService(map[string]string{user.Email: user.Email}, mockAuthzDB)
+
+	tkn, _ := auth.Issue(context.Background(), &mainflux.IssueReq{Id: user.ID, Email: user.Email, Type: 0})
+	token := tkn.GetValue()
+	userID, err := mainfluxSDK.CreateUser(token, user)
+	require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+	usertoken, err := mainfluxSDK.CreateToken(user)
+	require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+	user.ID = userID
+	user.Password = ""
+
+	cases := []struct {
+		desc     string
+		userID   string
+		token    string
+		err      error
+		response sdk.User
+	}{
+		{
+			desc:     "get existing user",
+			userID:   userID,
+			token:    usertoken,
+			err:      nil,
+			response: user,
+		},
+		{
+			desc:     "get non-existent user",
+			userID:   "43",
+			token:    usertoken,
+			err:      createError(sdk.ErrFailedFetch, http.StatusUnauthorized),
+			response: sdk.User{},
+		},
+
+		{
+			desc:     "get user with invalid token",
+			userID:   userID,
+			token:    wrongValue,
+			err:      createError(sdk.ErrFailedFetch, http.StatusUnauthorized),
+			response: sdk.User{},
+		},
+	}
+	for _, tc := range cases {
+		respUs, err := mainfluxSDK.User(tc.userID, tc.token)
+		assert.Equal(t, tc.err, err, fmt.Sprintf("%s: expected error %s, got %s", tc.desc, tc.err, err))
+		assert.Equal(t, tc.response, respUs, fmt.Sprintf("%s: expected response user %s, got %s", tc.desc, tc.response, respUs))
+	}
+}
+
+func TestUsers(t *testing.T) {
+	svc := newUserService()
+	ts := newUserServer(svc)
+	defer ts.Close()
+	sdkConf := sdk.Config{
+		UsersURL:        ts.URL,
+		MsgContentType:  contentType,
+		TLSVerification: false,
+	}
+
+	mainfluxSDK := sdk.NewSDK(sdkConf)
+	user := sdk.User{Email: "user@example.com", Password: "password"}
+
+	mockAuthzDB := map[string][]mocks.SubjectSet{}
+	mockAuthzDB[user.Email] = append(mockAuthzDB[user.Email], mocks.SubjectSet{Object: "authorities", Relation: "member"})
+	auth := mocks.NewAuthService(map[string]string{user.Email: user.Email}, mockAuthzDB)
+
+	tkn, _ := auth.Issue(context.Background(), &mainflux.IssueReq{Id: user.ID, Email: user.Email, Type: 0})
+	token := tkn.GetValue()
+
+	var users []sdk.User
+
+	for i := 1; i < 101; i++ {
+		email := fmt.Sprintf("test-%d@example.com", i)
+		password := fmt.Sprintf("password%d", i)
+		us := sdk.User{Email: email, Password: password, Metadata: metadata}
+		userID, err := mainfluxSDK.CreateUser(token, us)
+		require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+		us.ID = userID
+		us.Password = ""
+		users = append(users, us)
+	}
+	cases := []struct {
+		desc     string
+		token    string
+		offset   uint64
+		limit    uint64
+		err      error
+		response []sdk.User
+		name     string
+	}{
+		{
+			desc:     "get a list users",
+			token:    token,
+			offset:   0,
+			limit:    5,
```

Comment:
Define a constant for `limit`.

Reply:
The limit changes. As we test we can choose a limit of 5 or 110 for testing

Reply:
The limit is related to the size of the slice and it's reapeated many times. Creating a const you avoid any bug or repetition and it's self explanatory.